torch_xla/csrc/aten_xla_type.cpp

```diff
@@ -1591,10 +1591,16 @@ at::Tensor XLANativeFunctions::index(
   XLA_FN_COUNTER("xla::");
   CanonicalIndexInfo canonical_index_info =
       GetCanonicalIndexInfo(self, indices);
-  return bridge::AtenFromXlaTensor(
-      XLATensor::index(bridge::GetXlaTensor(canonical_index_info.base),
-                       bridge::GetXlaTensors(canonical_index_info.indices),
-                       canonical_index_info.start_dim));
+  c10::optional<Device> device =
+      bridge::GetXlaDevice(canonical_index_info.base);
+  if (!device.has_value()) {
+    device = bridge::GetXlaDevice(canonical_index_info.indices);
+  }
+  XLA_CHECK(device.has_value());
+  return bridge::AtenFromXlaTensor(XLATensor::index(
```

Comment:
Can we test this scenario? (i.e. having `base` and `index` on diff devices)

Reply:
Pytorch side has a test for this use case, https://github.com/pytorch/pytorch/pull/66506/files